### ISSUE TITLE
Averaging validation loss across steps

### ISSUE THREAD

**Issue by halfrot** [glorgao/SelectiveDPO#3]
Hi, thanks for the great work! I'm using this codebase to produce validation loss on my dataset to find difficult data points.

I have one question about the calculation of the validation loss [across different steps](https://github.com/glorgao/SelectiveDPO/blob/c1c7ded3154f6ce1ca55d136c66f3f3723e0bcd0/curriculum-lens/make_curricula.py#L24)
```
STEP_LIST = ['step0', 'step1', 'step2', 'step3', 'step4']
...
        for step in STEP_LIST:
...
```
I read through the paper on arXiv but didn’t find specific details on this. I had assumed the validation loss would be computed only using the best-performing reference model (i.e., the one with the lowest average validation loss). Does using all steps in this way influence the final results?

Thanks again for the codebase!

**Comment by glorgao** [comment]
Thank you for pointing out this important detail.

Firstly, the released code indeed computes the validation loss across multiple training steps. For example, with the UltraFeedback-Binarized dataset (~60k samples), calculation is performed about 5 times per epoch. For the smaller Argilla-DPO-Mix-7k dataset (~7k samples), we compute approximately 3-5 times per epoch (not so sure about the argila case, let me check later). 

Secondly, averaging the validation loss across multiple evaluation steps is the preferred approach. However, the results remain highly consistent: the Spearman correlation coefficient between (1) using only the last (we didn't test the case using best reference model) reference model and (2) averaging across 3-5 models ranges from 0.96 to 0.98. 

Below is a plot showing the evolution of the reward margin (defined as $\sigma$(reward margin)). It has stable and robust trends on the UltraFeedback-Binarized dataset.

<img width="1064" alt="Image" src="https://github.com/user-attachments/assets/397440dc-6f3d-4ee0-8f66-48d6e23333e4" />

One thing worthy note is that we don't trend to select the best reference model as the selecting is time-consuming.

**Comment by halfrot** [comment]
Thanks! It makes sense.

**Comment by glorgao** [comment]
Just a reminder: if you have any questions or encounter difficulties reproducing the curricula, benchmarking results, or hyperparameter tuning, please feel free to reach out. I'm happy to help and look forward to accelerating your progress :D
